src/init.cpp

```diff
@@ -0,0 +1,192 @@
+  DLAF_ASSERT(bool(handle_pool), "");
+  handle_pool.reset();
+}
+
+cublas::HandlePool getCublasHandlePool() {
+  DLAF_ASSERT(bool(handle_pool), "");
+  return *handle_pool;
+}
+
+template <>
+void Init<Backend::GPU>::initialize(configuration const& cfg) {
+  const int device = 0;
+  initializeNpCudaStreamPool(device, cfg.num_np_cuda_streams_per_thread);
+  initializeHpCudaStreamPool(device, cfg.num_hp_cuda_streams_per_thread);
+  initializeCublasHandlePool();
+  hpx::cuda::experimental::detail::register_polling(hpx::resource::get_thread_pool("default"));
+}
+
+template <>
+void Init<Backend::GPU>::finalize() {
+  finalizeNpCudaStreamPool();
+  finalizeHpCudaStreamPool();
+  finalizeCublasHandlePool();
+  hpx::cuda::experimental::detail::unregister_polling(hpx::resource::get_thread_pool("default"));
+}
+#endif
+
+template <typename T>
+void updateConfigurationValue(hpx::program_options::variables_map const& vm, T& var,
+                              std::string const& cmdline_option, std::string const& env_var);
+
+template <>
+void updateConfigurationValue(hpx::program_options::variables_map const& vm, std::size_t& var,
+                              std::string const& env_var, std::string const& cmdline_option) {
+  const std::string dlaf_env_var = "DLAF_" + env_var;
+  char* env_var_value = std::getenv(dlaf_env_var.c_str());
+  if (env_var_value) {
+    var = std::stoul(env_var_value);
+  }
+
+  const std::string dlaf_cmdline_option = "dlaf:" + cmdline_option;
+  if (vm.count(dlaf_cmdline_option)) {
+    var = vm[dlaf_cmdline_option].as<std::size_t>();
+  }
+}
+
+void updateConfiguration(hpx::program_options::variables_map const& vm, configuration& cfg) {
+  updateConfigurationValue(vm, cfg.num_np_cuda_streams_per_thread, "NUM_NP_CUDA_STREAMS_PER_THREAD",
+                           "num-np-cuda-streams-per-thread");
+  updateConfigurationValue(vm, cfg.num_hp_cuda_streams_per_thread, "NUM_HP_CUDA_STREAMS_PER_THREAD",
+                           "num-hp-cuda-streams-per-thread");
+}
+
+configuration& getConfiguration() {
+  static configuration cfg;
+  return cfg;
+}
+}
+
+hpx::program_options::options_description getOptionsDescription() {
+  hpx::program_options::options_description desc("DLA-Future options");
+
+  desc.add_options()("dlaf:help", "Print help message");
+  desc.add_options()("dlaf:print-config", "Print the DLA-Future configuration");
+  desc.add_options()("dlaf:num-np-cuda-streams-per-thread", hpx::program_options::value<std::size_t>(),
+                     "Number of normal priority CUDA streams per worker thread");
+  desc.add_options()("dlaf:num-hp-cuda-streams-per-thread", hpx::program_options::value<std::size_t>(),
+                     "Number of high priority CUDA streams per worker thread");
+
+  return desc;
+}
+
+void initialize(hpx::program_options::variables_map const& vm, configuration const& user_cfg) {
+  if (vm.count("dlaf:help") > 0) {
+    std::cout << getOptionsDescription() << std::endl;
```

Comment:
Sorry, I didn't test it, but it recalled me something from HPX: does it exit after "dlaf:help" or not?

Reply:
I think that the general expectation is that if any help is printed, then it stops without "running" the application.
But I don't know what's the best way of doing it together with HPX.

Reply:
Yes, agreed. I'll see what I can do here given the limitations of already running inside a task. As an aside, `--hpx:help` will already print the DLAF-specific options as well. I added this option to print only the DLAF-specific options. The former may be enough? I'll check what I can do in any case.

Reply:
Yes, exactly. I think it can be enough `--hpx:help`, even if it is not straightforward, it is what I do currently.